backend-rust/src/indexer.rs

```diff
@@ -505,13 +523,14 @@ impl ProcessEvent for BlockProcessor {
     }
 }
 
+#[derive(Clone)]
 struct BlockProcessingContext {
-    /// The last finalized block height according to the latest indexed block.
-    /// This is needed in order to compute the finalization time of blocks.
-    last_finalized_height: i64,
     /// The last finalized block hash according to the latest indexed block.
-    /// This is needed in order to compute the finalization time of blocks.
-    last_finalized_hash:   String,
+    /// This is used when computing the finalization time.
+    last_finalized_hash:  String,
+    /// The slot time of the last processed block.
+    /// This is used when computing the block time.
+    last_block_slot_time: NaiveDateTime,
```

Comment:
Why is it we do not use `DateTime<Utc>`?

Reply:
We don't store the timezone in the database, since they are all UTC, therefore [sqlx provides `NaiveDateTime`](https://docs.rs/sqlx/latest/sqlx/postgres/types/index.html#chrono) when querying and I we don't really gain much from converting it here.

Reply:
Oh I see, we are using `timestamp` instead of `timestamptz`! We should definitely change that. Using timestamp is definitely wrong - there is almost never any reason to use `timestamp` over `timestamptz`. Note that timestamptz does _not_ store the timezone, it just takes time zones into account when saving/loading data (the SQL type `timestamp with time zone` is very badly named). We definitely want to take it into account. For instance, if a PostgreSQL client changed their locale, the input time would be wrong.

Please change all instances of `timestamp` to `timestamptz` and use `DateTime<Utc>` instead (or I/you can do this in a separate PR).

Reply:
Let's fix this in a separate PR